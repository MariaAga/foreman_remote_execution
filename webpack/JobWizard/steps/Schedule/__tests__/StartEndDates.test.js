import React from 'react';
import * as patternfly from '@patternfly/react-core';
import { render, fireEvent, screen, act } from '@testing-library/react';
import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { StartEndDates } from '../StartEndDates';

jest.spyOn(patternfly, 'FormGroup');
patternfly.FormGroup.mockImplementation(props => <div props={props} />);

const setEnds = jest.fn();
const setIsNeverEnds = jest.fn();
const props = {
  starts: '',
  setStarts: jest.fn(),
  ends: 'some-end-date',
  setEnds,
  setIsNeverEnds,
  isNeverEnds: false,
};
const fixtures = {
  'renders isNeverEnds false': props,
  'renders isNeverEnds true': { ...props, isNeverEnds: true },
};

describe('StartEndDates', () => {
  testComponentSnapshotsWithFixtures(StartEndDates, fixtures);
  it('never ends', async () => {
    patternfly.FormGroup.mockRestore();
    await act(async () => render(<StartEndDates {...props} />));
    const neverEnds = screen.getByRole('checkbox', { name: 'Never ends' });
    await act(async () => fireEvent.click(neverEnds));
    expect(setIsNeverEnds).toBeCalledWith(true);
  });
});
