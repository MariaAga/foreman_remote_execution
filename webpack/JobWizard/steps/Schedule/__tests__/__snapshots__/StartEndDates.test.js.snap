// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartEndDates renders isNeverEnds false 1`] = `
<Fragment>
  <mockConstructor
    className="start-date"
    fieldId="start-date"
    label="Starts"
  >
    <DateTimePicker
      dateTime=""
      isDisabled={false}
      setDateTime={[MockFunction]}
    />
  </mockConstructor>
  <mockConstructor
    className="end-date"
    fieldId="end-date"
    helperTextInvalid="End time needs to be after start time"
    label="Ends"
    validated="success"
  >
    <DateTimePicker
      dateTime="some-end-date"
      isDisabled={false}
      setDateTime={[MockFunction]}
    />
    <Checkbox
      className=""
      id="never-ends"
      isChecked={false}
      isDisabled={false}
      isValid={true}
      label="Never ends"
      name="never-ends"
      onChange={[Function]}
    />
  </mockConstructor>
</Fragment>
`;

exports[`StartEndDates renders isNeverEnds true 1`] = `
<Fragment>
  <mockConstructor
    className="start-date"
    fieldId="start-date"
    label="Starts"
  >
    <DateTimePicker
      dateTime=""
      isDisabled={false}
      setDateTime={[MockFunction]}
    />
  </mockConstructor>
  <mockConstructor
    className="end-date"
    fieldId="end-date"
    helperTextInvalid="End time needs to be after start time"
    label="Ends"
    validated="success"
  >
    <DateTimePicker
      dateTime="some-end-date"
      isDisabled={true}
      setDateTime={[MockFunction]}
    />
    <Checkbox
      className=""
      id="never-ends"
      isChecked={true}
      isDisabled={false}
      isValid={true}
      label="Never ends"
      name="never-ends"
      onChange={[Function]}
    />
  </mockConstructor>
</Fragment>
`;
